trpcultivate_phenoshare/tests/src/Kernel/TripalImporter/ShareImporterFormTest.php

```diff
@@ -0,0 +1,753 @@
+
+    $this->assertEquals(
+      $expected['has_failed'],
+      $has_failed,
+      'The validation result array status does not match status returned by hasFailedValidation() method in scenario ' . $scenario
+    );
+  }
+
+  /**
+   * Data Provider: provides config value to 'allownew' configuration.
+   *
+   * If set to true, a trait can be inserted during file import, while a false
+   * value will notify user of a restriction.
+   *
+   * @return array
+   *   Each config value scenario is an array with the following values:
+   *   - A string, human-readable short title text of the scenario.
+   *   - Boolean, True to allow and False to restrict new trait during import.
+   *   - An array of expeceted values with the following keys.
+   *     - 'has_message': a false value will indicate that the form will post a
+   *       Drupal Status Message.
+   */
+  public function provideAllowNewConfig() {
+    return [
+      // #0: True, allow new traits to be added during import.
+      [
+        'allownew config set to true',
+        TRUE,
+        [
+          'has_message' => FALSE,
+        ],
+      ],
+
+      // #1: False, prevent traits from being added during import (notify).
+      [
+        'allownew config set to false',
+        FALSE,
+        [
+          'has_message' => TRUE,
+        ],
+      ],
+    ];
+  }
+
+  /**
+   * Test importer form notification relating to allow new config.
+   *
+   * @param string $scenario
+   *   A human-readable short title text of the scenario.
+   * @param bool $set_value
+   *   True to allow and False to restrict new trait during import.
+   * @param array $expected
+   *   An array of expeceted values with the following keys.
+   *     - 'has_message': a false value will indicate that the form will post a
+   *       Drupal Status Message.
+   *
+   * @dataProvider provideAllowNewConfig
+   */
+  public function testFormAllowNewNotification($scenario, $set_value, $expected) {
+
+    $form = $this->form_importer;
+    $form_state = new FormState();
+
+    $this->service_ConfigFactory
+      ->getEditable('trpcultivate_phenotypes.settings')
+      ->set('trpcultivate.phenotypes.ontology.allownew', $set_value)
+      ->save();
+
+    $this->phenoshare_importer->form($form, $form_state);
+
+    // Collect all messages of posted through the messenger service.
+    $messages = $this->service_Messenger->all();
+
+    $this->assertEquals(
+      $expected['has_message'],
+      isset($messages[$this->service_Messenger::TYPE_STATUS]),
+      'The form allow new notification message should coincide with the configured value ' . $scenario
+    );
+
+    if ($set_value == FALSE) {
+      $this->assertTrue(
+        str_contains((string) $messages[$this->service_Messenger::TYPE_STATUS][0], 'This module is set to NOT to allow new trait'),
+        'The status message text does not contain the expected message when allow new configuration is set to FALSE'
+      );
```

Comment:
This would be better done with a specialized assert like so:

```suggestion
      $this->assertStringContainsString(
        'This module is set to NOT to allow new trait',
        (string) $messages[$this->service_Messenger::TYPE_STATUS][0],
        'The status message text does not contain the expected message when allow new configuration is set to FALSE'
      );
```